art/defences/preprocessor/thermometer_encoding.py

```diff
@@ -146,6 +146,10 @@ def estimate_gradient(self, x: np.ndarray, grad: np.ndarray) -> np.ndarray:
         :param grad: Gradient value so far.
         :return: The gradient (estimate) of the defence.
         """
+        if self.channels_first:
+            x = np.swapaxes(x, 1, -1)
+            grad = np.swapaxes(grad, 1, -1)
```

Comment:
How about using `permute` which preserves the ordering of H and W instead of `swapaxes`?
```suggestion
        if self.channels_first:
            x = x.permute(0, 2, 3, 1)
            grad = grad.permute(0, 2, 3, 1)
```

Reply:
@beat-buesser What do you think about this: e.g.
```
```x = np.transpose(x, (0,) + tuple(range(2, len(x.shape))) + (1,))
```
That way it can be applied to inputs with extra dimensions, e.g. if someone wanted to use thermometer encoding for video data.

Reply:
@mathsinn I think that's a good idea, that should work.